org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java

```diff
@@ -397,16 +398,26 @@ private String constructTypeStub(ICompilationUnit parentCU, String name, int mod
 		if (modifiers != 0) {
 			buf.append(' ');
 		}
+
+		String[] permittedNames = fCompilationUnit.findPrimaryType().getPermittedSubtypeNames();
+		boolean isPermitted = Arrays.asList(permittedNames).stream().anyMatch(p -> name.equals(p));
+		if (isPermitted) {
+			buf.append("non-sealed ");
```

Comment:
I'd say the default would be `final`. Since the user is trying to restrict implementations, it'd make more sense to leave those restricted as well by default

Reply:
I don't remember if we're allowed to return snippets at that point, but if yes, and provided the client support snippets, then returning `${1:|final,non-sealed,sealed|}` would be the best.

Reply:
Cool feature. It would seem to be a great replacement for what JDT UI calls "linked proposals". There are proposals that allow cycling through a set of fixed values at various points of an applied code snippet/template.

From what I can tell, only CompletionItem supports snippets through the CompletionItem#setInsertTextFormat(InsertTextFormat.Snippet), assuming the client supports it. However, quick fixes/assists are represented as CodeAction, and I don't see a way of indicating the TextEdit(s) may contain snippets for resolution.